core/src/main/java/com/linecorp/armeria/server/HttpServerHandler.java

```diff
@@ -229,10 +229,17 @@ private void cleanup() {
         }
 
         if (!unfinishedRequests.isEmpty()) {
-            final ClosedSessionException cause = ClosedSessionException.get();
-            unfinishedRequests.forEach((req, res) -> {
+            final boolean cancel;
+            final Exception cause;
+            if (protocol.isMultiplex()) {
                 // An HTTP2 request is cancelled by Http2RequestDecoder.onRstStreamRead()
-                final boolean cancel = !protocol.isMultiplex();
+                cancel = false;
+                cause = ClosedStreamException.get();
```

Comment:
This method is called when a channel becomes inactive. So should we revert this?

Reply:
That sounds reasonable. 👍